### PR TITLE
chains Merkle shreds in broadcast fake shreds

### PR DESCRIPTION
**Pull request by behzadnouri**
#### Problem
Use chained Merkle shreds in `turbine/src/broadcast_stage/broadcast_fake_shreds_run.rs`.

#### Summary of Changes
Chains Merkle shreds in broadcast fake shreds.

### CODE REVIEW

**Review by carllin — APPROVED** [review]

File: turbine/src/broadcast_stage/broadcast_fake_shreds_run.rs [hunk 1]
```diff
@@ -45,6 +45,21 @@ impl BroadcastRun for BroadcastFakeShredsRun {
...
+            None => broadcast_utils::get_chained_merkle_root(
+                bank.slot(),
+                bank.parent_slot(),
+                blockstore,
+            )
+            .unwrap(),
+            Some(index) => {
+                let shred = blockstore
+                    .get_data_shred(bank.slot(), u64::from(index))
+                    .unwrap()
```
Comment (bw-solana) [inline turbine/src/broadcast_stage/broadcast_fake_shreds_run.rs:58]: Maybe use expect here?
```
.expect("Blockstore Error")
.expect("Shred not present")
```

File: turbine/src/broadcast_stage/broadcast_fake_shreds_run.rs [hunk 2]
```diff
@@ -45,6 +45,21 @@ impl BroadcastRun for BroadcastFakeShredsRun {
             .expect("Database error")
             .map(|meta| meta.consumed)
             .unwrap_or(0) as u32;
+        let chained_merkle_root = match next_shred_index.checked_sub(1) {
+            None => broadcast_utils::get_chained_merkle_root(
+                bank.slot(),
+                bank.parent_slot(),
+                blockstore,
+            )
+            .unwrap(),
```
Comment (bw-solana) [inline turbine/src/broadcast_stage/broadcast_fake_shreds_run.rs:54]: We have some nice error handling in `get_chained_merkle_root` but attempt to just unwrap it here. Maybe add an expect or something? Is there any way to fail gracefully in the event we can't get the chained merkle root?

**Review by bw-solana — COMMENTED** [review]
Looks fine to me. Curious if we could do more to handle any potential error cases

File: turbine/src/broadcast_stage/broadcast_fake_shreds_run.rs (hunk 1 above)
Comment (behzadnouri) [reply turbine/src/broadcast_stage/broadcast_fake_shreds_run.rs:58]: The 1st unwrap will print the `BlockstoreError` if it fails.
The 2nd unwrap will print `called Option::unwrap() on a None value`, indicating there was no shred.
Given that this code is only invoked in tests I think that is already sufficient here.

File: turbine/src/broadcast_stage/broadcast_fake_shreds_run.rs (hunk 2 above)
Comment (behzadnouri) [reply turbine/src/broadcast_stage/broadcast_fake_shreds_run.rs:54]: > Maybe add an expect or something?

The `unwrap` will actually print the error. That is why I am adding those new error types, which are descriptive of what went wrong. So not `expect` will add much more here. This is code is also invoked only in tests.

> Is there any way to fail gracefully in the event we can't get the chained merkle root?

This code is only invoked in tests and it fails we want to get a loud panic so to debug root cause.

**Review by carllin — APPROVED** [review]

**Pull request merged**